internal/transport/handler_server_test.go

```diff
@@ -392,7 +393,22 @@ func (s) TestHandlerTransport_HandleStreams_Timeout(t *testing.T) {
 			t.Errorf("ctx.Err = %v; want %v", err, context.DeadlineExceeded)
 			return
 		}
+		// rst flag setting to verify the noop function: signalDeadlineExceeded
+		ch := make(chan struct{}, 1)
+		origSignalDeadlineExceeded := signalDeadlineExceeded
+		signalDeadlineExceeded = func() {
+			ch <- struct{}{}
+		}
+		defer func() {
+			signalDeadlineExceeded = origSignalDeadlineExceeded
+		}()
 		s.WriteStatus(status.New(codes.DeadlineExceeded, "too slow"))
+		select {
```

Comment:
you have to wait on channel as well otherwise you are not verifying the if your function was triggered

```
select {
case <-ch:  // Signal received, continue with the test
case <-s.ctx.Done():
case <-time.After(5 * time.Second):
	t.Errorf("timeout waiting for ctx.Done")
	return
}
```

Reply:
done

Reply:
@misvivek i just saw you are doing this inside goroutine which won't affect the execution of test whether pass or fail.

You need to move both no op function setting and channel waiting code outside of go routine and check at the end when go runStream(s) is done, perhaps at the end after checking the trailers. So I would structure it like this 

```
// rst flag setting to verify the noop function: signalDeadlineExceeded
ch := make(chan struct{}, 1)
origSignalDeadlineExceeded := signalDeadlineExceeded
signalDeadlineExceeded = func() {
ch <- struct{}{}
}
defer func() {
signalDeadlineExceeded = origSignalDeadlineExceeded
}()

runStream := func(s *ServerStream) {....}

...
...

checkHeaderAndTrailer(t, rw, wantHeader, wantTrailer)
select {
case <-ch: // Signal received, continue with the test
case <-time.After(5 * time.Second):
t.Errorf("timeout waiting for ctx.Done")
return
}
```

Reply:
And i think the test will fail then because you haven't yet fixed the bug.